libs/api/src/main/java/de/cyface/api/model/ClassifiedSegment.java

```diff
@@ -0,0 +1,434 @@
+     *                        previous points.
+     * @param variance        The mathematical variance calculated from calibrated vertical accelerations.
+     * @param quality         The surface quality class calculated for this segment.
+     * @param dataPointCount  The number of sample points used to calculate {@code variance}. This is required to update
+     *                        {@code variance} without requiring previous points.
+     * @param version         The version of the format of this segment entry.
+     */
+    @SuppressWarnings("unused") // Part of the API
+    public ClassifiedSegment(final String oid, final boolean forward, final Geometry geometry, final double length,
+                             final Modality modality, final long vnk,
+                             final long nnk, final int start, final OffsetDateTime latestDataPoint, final String username,
+                             final double expectedValue, final double variance,
+                             final SurfaceQuality quality, final long dataPointCount, final String version) {
+        this.oid = oid;
+        this.forward = forward;
+        this.geometry = geometry;
+        this.length = length;
+        this.modality = modality;
+        this.vnk = vnk;
+        this.nnk = nnk;
+        this.start = start;
+        this.latestDataPoint = latestDataPoint;
+        this.username = username;
+        this.expectedValue = expectedValue;
+        this.variance = variance;
+        this.quality = quality;
+        this.dataPointCount = dataPointCount;
+        this.version = version;
+        Validate.isTrue(version.equals(CURRENT_VERSION));
+    }
+
+    /**
+     * Exports this measurement as GeoJSON feature.
+     *
+     * @param handler A handler that gets the GeoJson feature as string
+     */
+    public void asGeoJson(final Consumer<String> handler) {
+        // We decided to generate a String instead of using a JSON library to avoid dependencies in the model library
+
+        // measurement = geoJson "feature"
+        handler.accept("{");
+        handler.accept(jsonKeyValue("type", "Feature").getStringValue());
+        handler.accept(",");
+
+        // All tracks = geometry (MultiLineString)
+        handler.accept("\"geometry\":{");
+        handler.accept(jsonKeyValue("type", "MultiLineString").getStringValue());
+        handler.accept(",");
+        handler.accept("\"coordinates\":");
+        final var coordinates = convertToLineStringCoordinates(geometry);
+        handler.accept(coordinates);
+        handler.accept("},");
+
+        final var oid = jsonKeyValue("oid", getOid()); // for support
+        final var wayId = jsonKeyValue("@id", "way/232814001"); // FIXME: missing
+        final var highwayType = jsonKeyValue("highway", "residential"); // FIXME: missing
+        final var surfaceType = jsonKeyValue("surface", "asphalt"); // FIXME: missing
```

Comment:
Before, we also displayed the OSM way id and in some cases (Datenkost Mockup) the OSM tags "highway type" and "surface type" in the web app.

Should we create a task to annotate this data in the surface pipeline and store it into the segments or should we wait until we need this in a use-case?

Reply:
[v](https://cyface.atlassian.net/browse/DAT-966)